File: stepik-java-api/src/main/java/org/stepik/api/queries/sections/StepikSectionsGetQuery.java
package org.stepik.api.queries.sections;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.sections.Sections;
import org.stepik.api.queries.StepikAbstractGetQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikSectionsGetQuery extends StepikAbstractGetQuery<StepikSectionsGetQuery, Sections> {
    public StepikSectionsGetQuery(@NotNull StepikAbstractAction stepikAction) {
        super(stepikAction, Sections.class);
    }

    @NotNull
    public StepikSectionsGetQuery page(int value) {
        addParam("page", value);
        return this;
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.SECTIONS;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/sections/StepikSectionsGetQuery.kt
package org.stepik.api.queries.sections

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.sections.Sections
import org.stepik.api.queries.StepikAbstractGetQuery

class StepikSectionsGetQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractGetQuery<StepikSectionsGetQuery, Sections>(stepikAction, Sections::class.java) {
    
    fun page(value: Int): StepikSectionsGetQuery {
        addParam("page", value)
        return this
    }
    
    override val url = "${stepikAction.stepikApiClient.host}/api/sections"
    
}
